.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,56 @@
+on:
+  push:
+    branches: [ main, mt/docker-and-aws ]
+  pull_request:
+    branches: [ main ]
```

Comment:
Doporucoval bych ten build rozdelit na dve casti a pridat podminku, kdyz se neco zmeni v danem adresari. At  se to nebuildi zbytecne. Taky bych ocekaval, ze ten build je jenom pri push do mainu, nikdy jindy. Ve finale napriklad takhle:
```
on:
  push:
    branches:
      - main
    paths:
      - 'backend/**'
 ```     
 A potom  bych vytvoril druhy file s buildem pipeline pro:
 ```
 on:
  push:
    branches:
      - main
    paths:
      - 'pipeline/**'
  ```

Reply:
Ok, rozdelil jsem je.